tests/test.py

```diff
@@ -7,11 +7,56 @@
 
 from boto.s3.connection import S3Connection
 from boto.s3.key import Key
+import dcos.config
+import dcos.http
+import dcos.package
 import os
 import pytest
 import re
 import shakedown
 import subprocess
+import urllib
+
+
+def setup_module(module):
+    _require_package('hdfs')
+    _install_spark()
+
+
+def _install_spark():
+    options = {"hdfs":
+               {"config-url":
+                "http://hdfs.marathon.mesos:9000/v1/connection"}}
+
+    if os.environ.get('SECURITY') == 'strict':
+        options['service'] = {"user": "nobody",
+                              "principal": "service-acct",
+                              "secret_name": "secret" }
+
+    shakedown.install_package('spark', options_json=options, wait_for_completion=True)
+
+    def pred():
+        dcos_url = dcos.config.get_config_val("core.dcos_url")
+        spark_url = urllib.parse.urljoin(dcos_url, "/service/spark")
+        status_code = dcos.http.get(spark_url).status_code
+        return status_code == 200
```

Comment:
OK. Should we add in a check to make sure all the HDFS servers (JournalNodes, NameNodes, ZKFCs, DataNodes) are up before launching the Spark job? We do that in the HDFS integration tests.